### ISSUE TITLE
Call to an undefined method BelongsToMany::withTrashed()

### ISSUE THREAD

**Issue by matro-skin** [laravel/framework#32487]
- Laravel Version: 5.7
- PHP Version: 7.2
- Database Driver & Version: MySQL 5.7

### Description:
I use relations like this
```
public function experts()
{
    return $this->belongsToMany(Expert::class, 'app_experts', 'app_id', 'expert_id')
                ->withTrashed();
}
```
All is ok, but php artisan code:analyse tells me that:
`Call to an undefined method Illuminate\Database\Eloquent\Relations\BelongsToMany::withTrashed()`
What's wrong with it?

If is's important `Expert` model use `SoftDeletes` trait.

**Comment by DrCake** [comment]
I'm guessing you're using LaraStan which used to provided that `code:analyse` artisan command. Might be best to check their first as it sounds like the issue is only related to that and not the behaviour of the code. If you are using LaraStan, try updating the package as they removed that command a while ago and this issue may be fixed in a more current version.

**Comment by matro-skin** [comment]
@DrCake, Bytheway IDE doesn't recognize this method too...

**Comment by DrCake** [comment]
> @DrCake, Bytheway IDE doesn't recognize this method too...

That's because the call to `withTrashed()` is intercepted by the php magic call method and is called as a scope on the found `Expert` model. IDE's and static code analysers have a hard time understanding this as they would probably have to actually execute the code to know it works. When exectuted the code should work fine though.

Looks like this was fixed in LaraStan version 0.4.2 by [this PR](https://github.com/nunomaduro/larastan/pull/319)

You might also want to check out the  [ide-helper](https://github.com/barryvdh/laravel-ide-helper) package which helps the IDE recognise a lot of the 'magic' in laravel.

**Comment by driesvints** [comment]
Hey there,

Unfortunately we don't support this version anymore. Please check out [our support policy](https://laravel.com/docs/releases) on which versions we are currently supporting. Can you please try to upgrade to the latest version and see if your problem persists? We'll help you out and re-open this issue if so.

Thanks!
